app/src/org/commcare/dalvik/activities/LoginActivity.java

```diff
@@ -311,24 +326,29 @@ protected void onResume() {
         
         try {
             //TODO: there is a weird circumstance where we're logging in somewhere else and this gets locked.
-        if(CommCareApplication._().getSession().isActive() && CommCareApplication._().getSession().getLoggedInUser() != null) {
-            Intent i = new Intent();
-            i.putExtra(ALREADY_LOGGED_IN, true);
-            setResult(RESULT_OK, i);
-            
-            CommCareApplication._().clearNotifications(NOTIFICATION_MESSAGE_LOGIN);
-            finish();
-            return;
+            if (CommCareApplication._().getSession().isActive() && CommCareApplication._().getSession().getLoggedInUser() != null) {
+                Intent i = new Intent();
+                i.putExtra(ALREADY_LOGGED_IN, true);
+                setResult(RESULT_OK, i);
+                
+                CommCareApplication._().clearNotifications(NOTIFICATION_MESSAGE_LOGIN);
+                finish();
+                return;
+            }
+        } catch (SessionUnavailableException sue) {
+            // Nothing, we're logging in here anyway
         }
-        }catch(SessionUnavailableException sue) {
-            //Nothing, we're logging in here anyway
+
+        //If we arrived at LoginActivity from clicking the regular app icon, and there
+        //are no longer any available apps, we want to redirect to CCHomeActivity
+        if (CommCareApplication._().getReadyAppRecords().size() == 0) {

```

Comment:
Maybe... The thing is that we get things like Localizations from the seated app, so I think even though the login should allow the seated app to change, it still will need to know that an app is seated to be able to display things in the right language/icon set/etc.